src/test/java/core/basesyntax/HelloWorldTest.java

```diff
@@ -1,8 +1,145 @@
 package core.basesyntax;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.exceptions.RegistrationFailureException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
 /**
  * Feel free to remove this class and create your own.
  */
 public class HelloWorldTest {
+    private static RegistrationService registrationService;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    void register_addUser_Ok() {
+        User user = new User();
+        user.setLogin("Marco123");
+        user.setPassword("12345678");
+        user.setAge(20);
+
+        User result = registrationService.register(user);
+        User storageUser = registrationService.getUser(user.getLogin());
+        assertEquals(storageUser, result);
+    }
+
+    @Test
+    void register_passwordEdges_Ok() {
+        User user1 = new User();
+        user1.setLogin("Marco1");
+        user1.setPassword("123456");
+        user1.setAge(27);
+
+        User result = registrationService.register(user1);
+        assertEquals(result, user1);
+
+        User user2 = new User();
+        user2.setLogin("Parmezan");
+        user2.setPassword("12345678910");
+        user2.setAge(50);
+        result = registrationService.register(user2);
+        assertEquals(result, user2);
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        User user = null;
+        assertThrows(RegistrationFailureException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_addIdenticalUsers_notOk() {
+        User user = new User();
+        user.setLogin("Antonio");
+        user.setPassword("12345678");
+        user.setAge(23);
```

Comment:
couple of moments here:
* there is not need to create a new user and populate it with data by hand each time, add a `validUser` field  in the test class, and initialize it with the valid fields `BeforeEach` test. Then when you need to assert a validation error, just set a single not valid field to it
* magic values should be constants, same for each other test method